### PR TITLE
[Fabric] Initial support to host React Native as a ContentIsland

### PR DESCRIPTION
**Pull request by acoates-ms**
## Description
- Added support for hosting a `CompositionRootView` in a `ContentIsland`.
- Updated Input objects to work with WinAppSDK input objects
- Fixed windows message-based Input object to provide coordinated in scaled coordinates.
- Switched the playground app to be able to run either within a WinAppSDK `ContentIsland` using WinAppSDK composition or using system composition.  This is controlled by a new checkbox in the settings dialog.
 - Switched the `ScaleFactor` properties in the ABI from `double` to `float` to align with `ContentIsland`s and other places exposing display scale factor.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12227)

### CODE REVIEW

**Review by vmoroz — APPROVED** [review]

**Review by marlenecota — APPROVED** [review]

File: packages/playground/windows/playground-composition/Playground-Composition.cpp [hunk 1]
```diff
@@ -273,13 +166,83 @@ struct WindowData {
...
+              auto site = bridge.Site();
+              auto siteWindow = site.Environment();
+              auto displayScale = siteWindow.DisplayScale();
+
+              site.ParentScale(displayScale);
+              site.ActualSize({metrics.Frame.Width / 2, metrics.Frame.Height / 2});
+              site.ClientSize(
+                  {static_cast<int32_t>(metrics.Frame.Width / 2 * metrics.PointScaleFactor),
+                  static_cast<int32_t>(metrics.Frame.Height / 2 * metrics.PointScaleFactor)});
+            */
```
Comment (chrisglein) [inline packages/playground/windows/playground-composition/Playground-Composition.cpp:209]: Is there a tracking item in WinUI's repo/roadmap that we can add a link to here?

File: packages/playground/windows/playground-composition/Playground-Composition.cpp [hunk 2]
```diff
@@ -273,13 +166,83 @@ struct WindowData {
...
+
+            // bridge.OverrideScale(ScaleFactor(hwnd));
```
Comment (chrisglein) [inline packages/playground/windows/playground-composition/Playground-Composition.cpp:211]: You've got a couple commented out lines here (211 and 216). Delete?

File: packages/playground/windows/playground-composition/Playground-Composition.cpp [hunk 3]
```diff
@@ -311,9 +274,29 @@ struct WindowData {
     return 0;
   }
 
-  LRESULT TranslateMessage(UINT message, WPARAM wparam, LPARAM lparam) noexcept {
-    if (m_CompositionHwndHost) {
-      return static_cast<LRESULT>(m_CompositionHwndHost.TranslateMessage(message, wparam, lparam));
+  float ScaleFactor(HWND hwnd) noexcept {
+    return GetDpiForWindow(hwnd) / 96.0f;
```
Comment (chrisglein) [inline packages/playground/windows/playground-composition/Playground-Composition.cpp:278]: I _think_ this (96) is defined in a header as `USER_DEFAULT_SCREEN_DPI`

File: packages/playground/windows/playground-composition/Playground-Composition.cpp [hunk 4]
```diff
@@ -530,6 +513,8 @@ int RunPlayground(int showCmd, bool useWebDebugger) {
 
   WINRT_VERIFY(hwnd);
 
+  g_hwndTopLevel = hwnd; // Temporary for prototyping
```
Comment (chrisglein) [inline packages/playground/windows/playground-composition/Playground-Composition.cpp:516]: Plan for removal?

File: packages/playground/windows/playground-composition/Playground-Composition.cpp [hunk 5]
```diff
@@ -539,11 +524,21 @@ int RunPlayground(int showCmd, bool useWebDebugger) {
   HACCEL hAccelTable = LoadAccelerators(WindowData::s_instance, MAKEINTRESOURCE(IDC_PLAYGROUND_COMPOSITION));
 
   MSG msg = {};
+
+  // This would be the same as the loop below. - Using the lower loop right now for easier debugging.
+  // g_liftedDispatcherQueueController.DispatcherQueue().RunEventLoop();
```
Comment (chrisglein) [inline packages/playground/windows/playground-composition/Playground-Composition.cpp:529]: When to remove?

**Pull request merged**